### DISCUSSION TITLE
Using example in Rehearse Function

### DISCUSSION THREAD

**Discussion by NaziaToma** [explosion/spaCy#8714] [Help: Coding & Implementations] [answered]
How should I use examples in nlp.rehearse(examples, sgd=optimizer)?

My code: 
for text, annotations in revision_data["annotations"]:
        # create Example
    doc = nlp.make_doc(text)
    example = Example.from_dict(doc, annotations)
    #split_examples = example.split_sents()
nlp.rehearse(example, sgd=optimizer)

Error: 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_19992/1192699014.py in <module>
      4     example = Example.from_dict(doc, annotations)
      5     #split_examples = example.split_sents()
----> 6 nlp.rehearse(example, sgd=optimizer)

~\miniconda3\lib\site-packages\spacy\language.py in rehearse(self, examples, sgd, losses, component_cfg, exclude)
   1155         if len(examples) == 0:
   1156             return
-> 1157         validate_examples(examples, "Language.rehearse")
   1158         if sgd is None:
   1159             if self._optimizer is None:

~\miniconda3\lib\site-packages\spacy\training\example.pyx in spacy.training.example.validate_examples()

TypeError: [E978] The Language.rehearse method takes a list of Example objects, but got: <class 'spacy.training.example.Example'>

**Comment by adrianeboyd** [comment, marked as answer]
For this specific error the argument should be `List[Example]` not `Example`, so `[example]`.

The rehearse functionality hasn't been well tested with v3 yet, so you still may run into errors or unexpected results.

**Reply by NaziaToma** [reply]
I am getting this error now TypeError: unhashable type: 'numpy.ndarray'

**Reply by adrianeboyd** [reply]
Yeah, this may not work currently, sorry!
